File: 007_reverse_int.py
'''
Link: https://leetcode.com/problems/reverse-integer/
'''
import math


class Solution:
    def reverse(self, x: int) -> int:

        lower_limit = -math.pow(2, 31)
        upper_limit = math.pow(2, 31)-1

        number = str(x)
        if (x < 0):
            number = number[1:]

        number = number[::-1]
        number = int(number)

        if (lower_limit <= number <= upper_limit):

            if (x < 0):
                number *= -1

            return number

        return 0

File: 007_reverse_int.ts
function _rev(x: string) {
    return x.split("").reverse().join("");
}

function reverse(x: number): number {
    const lowerLimit = -Math.pow(2, 31);
    const upperLimit = Math.pow(2, 31) - 1;
    let xStr = x.toString();

    if (x < 0) {
        xStr = xStr.substring(1);
    }

    let ans = Number.parseInt(_rev(xStr));

    if (x < 0) {
        ans *= -1;
    }

    if (ans < lowerLimit || ans > upperLimit) return 0;

    return ans;
}
